### PR TITLE
[backport-110] SCA hardening for modular field inversion in EC_GROUPs

### PR DESCRIPTION
**Pull request by romen**
Backport of #8254 to `1.1.0-stable`.

Cherry-picking required manual changes in:
- `CHANGES`
- `crypto/ec/ecp_smpl.c` (`BN_rand_range` instead of `BN_priv_rand_range`)
- `crypto/ec/ec2_smpl.c` (`EC_GF2m_simple_method()` moved to the bottom of the file)
- `crypto/ec/ec_err.c`, `include/openssl/ec.h` (`make update`)
- 7d9a017 manually makes sure that error codes match `1.1.1-stable`/`master`

The tests required a major rewrite as we don't have `libtestutil.a` in `1.1.0-stable`.

## Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/ec_internal_test.c [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+ */
+
+#include <openssl/ec.h>
+#include "ec_lcl.h"
+#include <openssl/objects.h>
+
+#include "testutil.h"
+
+#define TEST_info(...) \
+    fprintf(stdout, __VA_ARGS__)
```
Comment (mattcaswell) [inline test/ec_internal_test.c]: This line is causing travis to fail. Apparently this is a C99 feature.

````
ccache gcc  -I. -Iinclude -Icrypto/ec -Icrypto/include -DDSO_DLFCN -DHAVE_DLFCN_H -DNDEBUG -DOPENSSL_THREADS -DOPENSSL_NO_DYNAMIC_ENGINE -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DRC4_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DPADLOCK_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib/engines-1.1\"" -Wall -O3 -pthread -m64 -DL_ENDIAN  -DDEBUG_UNUSED -DPEDANTIC -pedantic -Wno-long-long -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wsign-compare -Wmissing-prototypes -Wshadow -Wformat -Wtype-limits -Wundef -Werror -Wa,--noexecstack  -MMD -MF test/ec_internal_test.d.tmp -MT test/ec_internal_test.o -c -o test/ec_internal_test.o test/ec_internal_test.c
test/ec_internal_test.c:16:19: error: anonymous variadic macros were introduced in C99 [-Werror=variadic-macros]
 #define TEST_info(...) \
                   ^
cc1: all warnings being treated as errors
make[1]: *** [test/ec_internal_test.o] Error 1
make[1]: Leaving directory `/home/travis/build/openssl/openssl'
make: *** [build_programs] Error 2
````

Comment (romen) [reply test/ec_internal_test.c]: right! replaced `TEST_info()` directly with `fprintf`.

**Comment by romen** [comment]
I also updated the copyright years of changed files.

**Comment by romen** [comment]
Merged to `1.1.0-stable`:
- 09c11fe59b3d45d35e61d95d0f3a5a371f96a19d
- b7fc0784c4cfe81db8728f814925c6f98dd948d1

**Pull request closed without merging**